### PR TITLE
Add gifs for use cases in Usage

### PR DESCRIPTION
**Pull request by NH0**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -29,3 +29,6 @@ build/
 
 ### VS Code ###
 .vscode/
+
+### MAC OS ###
+*.DS_Store
```
Comment (wrousseau) [inline .gitignore]: à ne pas mettre dans le projet

File: README.md [hunk 2]
```diff
@@ -74,8 +74,6 @@ Please note that providing such a feature on your API does not come without risk
 
 * [Kotlin](https://kotlinlang.org/)
 * [Spring Boot](https://spring.io/projects/spring-boot)
-* []()
```
Comment (wrousseau) [inline README.md]: pas d'autres librairies utilisées? notamment pour le parser lexer

File: README.md [hunk 3]
```diff
@@ -115,42 +113,93 @@ mvn install #inside the spring-search folder
     <artifactId>spring-search</artifactId>
     <version>0.0.4-SNAPSHOT</version>
 </dependency>
+```
+
+
```
Comment (wrousseau) [inline README.md]: saute 2 lignes

File: README.md [hunk 4]
```diff
@@ -115,42 +113,93 @@ mvn install #inside the spring-search folder
...
+<!-- USAGE EXAMPLES -->
+## Usage
 
+Create a Rest repository extending JpaSpecificationExecutor
```
Comment (wrousseau) [inline README.md]: Your repository should be annotated as a `RepositoryRestResource` and should extend `JpaSpecificationExecutor`

File: README.md [hunk 5]
```diff
@@ -115,42 +113,93 @@ mvn install #inside the spring-search folder
...
 ```
-5. Import the library in your controller
+import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
+import org.springframework.data.rest.core.annotation.RepositoryRestResource;
+
+@RepositoryRestResource
+public interface yourRepository extends JpaRepository<yourModel, Integer>, JpaSpecificationExecutor<yourModel> {
```
Comment (wrousseau) [inline README.md]: class names, even example ones should start with an uppercase letter

File: README.md [hunk 6]
```diff
@@ -115,42 +113,93 @@ mvn install #inside the spring-search folder
...
+Request : `/cars?search=color:Red OR color:Blue`  
+![or operator example](./docs/images/or-example.gif)
+
+7. Using the `AND` operator  
+Request : `/cars?search=brand:Aston* AND price<300000`  
+![and operator example](./docs/images/and-example.gif)
+
+8. Using parenthesis  
+Request : `/cars?search=( brand:Nissan OR brand:Chevrolet ) AND color:Blue`  
+*Note: Space inside parenthesis is not necessary*
```
Comment (wrousseau) [inline README.md]: Spaces inside the parenthesis are not

File: README.md [hunk 7]
```diff
@@ -115,42 +113,93 @@ mvn install #inside the spring-search folder
...
+Request : `/cars?search=model:'Spacetourer Business Lounge'`  
+![space example](./docs/images/space-example.gif)
 
-_For more examples, please refer to the [Documentation](https://example.com)_
+10. Using special characters  
+Request: `/cars?search=model:中华V7`
+![special characters example](./docs/images/special-characters-example.gif)
 
+11. Using deep fields  
+Request : `/cars?search=options.transmission:Auto`
```
Comment (wrousseau) [inline README.md]: You don't give a warning about deep fields?

File: README.md [hunk 8]
```diff
@@ -115,42 +113,93 @@ mvn install #inside the spring-search folder
...
+![deep field example](./docs/images/deep-field-example.gif)
 
+12. Full example
```
Comment (wrousseau) [inline README.md]: Complex example

File: README.md [hunk 9]
```diff
@@ -115,42 +113,93 @@ mvn install #inside the spring-search folder
...
+12. Full example  
+Request : `/cars?search=creationyear:2018 AND price<300000 AND (color:Yellow OR color:Blue) AND options.transmission:Auto`
+![full example](./docs/images/complete-example.gif)
 
 <!-- ROADMAP -->
 ## Roadmap
 
 See the [open issues](https://github.com/sipios/spring-search/issues) for a list of proposed features (and known issues).
 
+Please note that boolean parameter types isn't yet supported.
```
Comment (wrousseau) [inline README.md]: are yet to be supported

File: README.md (hunk 7 above)
Comment (NH0) [reply README.md]: No, they work well on strings

**Pull request merged**